### PR TITLE
Require "edit submissions", not "add submissions", for working with transcripts and translations

### PR DESCRIPTION
**Pull request by jnm**
## Checklist

1. [x] If you've added code that should be tested, add tests
2. [ ] If you've changed APIs, update (or create!) the documentation
3. [x] Ensure the tests pass
4. [x] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
5. [x] Write a title and, if necessary, a description of your work suitable for publishing in our [release notes](https://community.kobotoolbox.org/tag/release-notes)
6. [ ] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes

## Description

Doing any kind of work with transcripts or translations—whether adding a new one or editing an existing one—now always requires the "edit submissions" permission.

### CODE REVIEW

File: kobo/apps/subsequences/api_view.py [hunk 1]
```diff
@@ -48,8 +49,17 @@ def _check_asr_mt_access_if_applicable(user, posted_data):
...
 
 
+class AdvancedSubmissionPermission(SubmissionPermission):
+    """
+    Regular `SubmissionPermission` maps POST to `add_submissions`, but
+    `change_submissions` should be required here
+    """
+    perms_map = deepcopy(SubmissionPermission.perms_map)
+    perms_map['POST'] = ['%(app_label)s.change_%(model_name)s']
+
```
Comment (noliveleger) [inline kobo/apps/subsequences/api_view.py:59]: I know if only few lines but I would have preferred to keep the permission class in its own file (to be consistent with the rest of the app).

File: kobo/apps/subsequences/tests/test_submission_extras_api_post.py [hunk 2]
```diff
@@ -88,6 +95,115 @@ def test_translation_revisions_stored_properly(self):
...
+        resp = self.client.post(schema['url'], modified, format='json')
+        assert resp.status_code == 404
+
+        self.asset.assign_perm(other_user, PERM_CHANGE_ASSET)
+        resp = self.client.post(schema['url'], modified, format='json')
+        assert resp.status_code == 404
+
+        self.asset.assign_perm(other_user, PERM_VIEW_SUBMISSIONS)
+        resp = self.client.post(schema['url'], modified, format='json')
+        assert resp.status_code == 403
```
Comment (noliveleger) [inline kobo/apps/subsequences/tests/test_submission_extras_api_post.py:172]: Not related to this PR but I wondering why `PERM_VIEW_ASSET`, `PERM_CHANGE_ASSET` return a 404 while `PERM_VIEW_SUBMISSIONS` returns a 403. I was assuming that as soon as user had `PERM_VIEW_ASSET`, users would receive a 403 when they don't have required permissions on nested objects.

**Review by noliveleger — COMMENTED** [review]
PR is ok and LGTM. 
Just have comments / questions before merging.  Feel free to merge it if my comments are not relevant.

File: kobo/apps/subsequences/tests/test_submission_extras_api_post.py (hunk 2 above)
Comment (jnm) [reply kobo/apps/subsequences/tests/test_submission_extras_api_post.py:172]: Good question; I'd assume it's just the usual DRF behavior of:
* Did you send a GET for a specific object that you are not allowed to see? We're not telling you whether or not it exists, so 404
* Did you try to take a disallowed action (POST, PATCH) on an object that you *are* allowed to see? → 403

I didn't expect nested objects to behave differently, but we can look into it more if you think something doesn't make sense

File: kobo/apps/subsequences/api_view.py (hunk 1 above)
Comment (jnm) [reply kobo/apps/subsequences/api_view.py:59]: Yeah, I thought about it. I didn't want to make a tiny file in the `subsequences` app, and I wasn't sure about putting it in the same file as `SubmissionPermission`, where the name "advanced" might be confusing to read without the context of `subsequences`. For best readability, I would just leave it here next to the only thing that uses it, but if more extensive `subsequences` permission work is needed in the future, or the permissions class gets used in other places, then I'd be in favor of moving to a separate file.

File: kobo/apps/subsequences/tests/test_submission_extras_api_post.py (hunk 2 above)
Comment (noliveleger) [reply kobo/apps/subsequences/tests/test_submission_extras_api_post.py:172]: Right! that's what I'm expecting too. 

I missed [an important line](https://github.com/kobotoolbox/kpi/blob/require-change-not-add-for-transcripts-translations/kpi/permissions.py#L343)! 

I was thinking the model was `Asset`.  Everything is good.

**Review by noliveleger — APPROVED** [review]

**Pull request merged**